core/src/test/scala/org/apache/spark/storage/DiskBlockManagerSuite.scala

```diff
@@ -141,6 +141,20 @@ class DiskBlockManagerSuite extends SparkFunSuite with BeforeAndAfterEach with B
     assert(attemptId.equals("1"))
   }
 
+  test("SPARK-37618: Sub dirs are group writable") {
+    val conf = testConf.clone
+    conf.set("spark.local.dir", rootDirs)
+    conf.set("spark.shuffle.service.enabled", "true")
+    conf.set("spark.shuffle.service.removeShufle", "true")
```

Comment:
We need to a negative test case whose config is `spark.shuffle.service.removeShufle`.

According to the AS-IS PR, we don't have a test coverage because the default was `true`.